src/math/math_base.h

```diff
@@ -105,5 +106,65 @@ struct PointBase3D : public PointBase2D<_Type>
     _Type z;
 };
 
+template <typename _Type>
+struct LineBase2D
+{
+    LineBase2D( const PointBase2D<_Type> & _p1, const PointBase2D<_Type> & _p2 )
+        : p1( _p1 ), p2( _p2 )
+    { }
+
+    LineBase2D( const PointBase2D<_Type> & _p, double angleRadians )
+        : p1( _p ), p2( _p.x + std::cos(angleRadians), _p.y + std::sin(angleRadians) )
+    { }
+
+    bool operator == ( const LineBase2D & line ) const
+    {
+        return p1 == line.p1 && p2 == line.p2;
+    }
+
+    // This is translation (shift) function
+    LineBase2D & operator += ( const PointBase2D<_Type> & point )
```

Comment:
This code is wrong. We create a temporary object inside function and return a reference to it after exiting the function.

If we want to implement translation function we need to use operator **+** and return a reference to `*this`.